runtime/compiler/trj9/optimizer/JProfilingValue.cpp

```diff
@@ -280,15 +280,42 @@ TR_JProfilingValue::lowerCalls()
 
          dumpOptDetails(comp(), "%s Replacing profiling placeholder n%dn with value profiling trees\n",
             optDetailString(), child->getGlobalIndex());
-
+         TR::TreeTop *prevTT = cursor->getPrevTreeTop();
+         TR::TreeTop *nextTT = cursor->getNextTreeTop();
+         TR::Node *prevTTNode = prevTT->getNode();
          // Extract the arguments and add the profiling trees
          TR::Node *value = child->getFirstChild();
          TR_AbstractHashTableProfilerInfo *table = (TR_AbstractHashTableProfilerInfo*) child->getSecondChild()->getAddress();
+         // While lowering value profiling calls, Scan the previous trees with the next trees to make sure by lowering the value profiler calls,
+         // We are not splitting blocks which orphans compressedref anchor nodes.
+         if (prevTTNode->getNumChildren() >=1 && prevTTNode->getFirstChild()->getType() == TR::Address)
+            {
+            for (TR::TreeTop *itr = nextTT; itr && 
+               (itr->getNode()->getOpCodeValue() != TR::BBStart || itr->getNode()->getBlock()->isExtensionOfPreviousBlock());
+               itr = itr->getNextTreeTop())
+               {
+               if (itr->getNode()->getOpCodeValue() == TR::compressedRefs
+                  && itr->getNode()->getFirstChild() == prevTTNode->getFirstChild())
+                  {
+                  traceMsg(comp(), "Moving treetop node n%dn above the profiling call to avoid uncommoning\n", itr->getNode()->getGlobalIndex());
+                  TR::TreeTop *itrPrev = itr->getPrevTreeTop();
+                  TR::TreeTop *itrNext = itr->getNextTreeTop();
+                  prevTT->setNextTreeTop(itr);
+                  cursor->setPrevTreeTop(itr);
+                  itr->setPrevTreeTop(prevTT);
+                  itr->setNextTreeTop(cursor);
+                  itrPrev->setNextTreeTop(itrNext);
+                  itrNext->setPrevTreeTop(itrPrev);
```

Comment:
Are we able to use one of the following APIs to simplify this code greatly?

https://github.com/eclipse/omr/blob/f698070a815a92a8f1da548b6de042ef73bcd9e7/compiler/il/OMRTreeTop_inlines.hpp#L80-L103

Reply:
Those simplified inlines should be used here. Let me make changes. Thanks for the suggestion...